File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/request/BusinessRequest.java
package com.ll.hotel.domain.member.member.dto.request;

import jakarta.validation.constraints.*;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.List;
import java.util.Map;

public class BusinessRequest {
        public record RegistrationInfo(
                @NotBlank(message = "사업자 등록 번호는 필수 항목입니다.")
                @Pattern(regexp = "^[0-9]{10}$", message = "사업자 등록 번호는 10자리 숫자여야 합니다.")
                String businessRegistrationNumber,

                @NotNull(message = "개업 일자는 필수 항목입니다.")
                @PastOrPresent(message = "개업 일자는 현재 날짜 또는 과거여야 합니다.")
                LocalDate startDate,

                @NotBlank(message = "대표자명은 필수 항목입니다.")
                @Size(max = 30, message = "대표자명은 최대 30자까지 가능합니다.")
                String ownerName
        ) {}

        public record RegistrationApiForm(
                List<Map<String, String>> businesses
        ) {
                public static RegistrationApiForm from(RegistrationInfo registrationInfo) {
                        return new RegistrationApiForm(List.of(Map.of(
                                "b_no", registrationInfo.businessRegistrationNumber(),
                                "start_dt", registrationInfo.startDate().format(DateTimeFormatter.ofPattern("yyyyMMdd")),
                                "p_nm", registrationInfo.ownerName()
                        )));
                }
        }
}

File: backend/src/main/java/com/ll/hotel/domain/member/member/dto/request/BusinessRequest.kt
package com.ll.hotel.domain.member.member.dto.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PastOrPresent
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size
import java.time.LocalDate
import java.time.format.DateTimeFormatter

class BusinessRequest {
    data class RegistrationInfo(
        @field:NotBlank
        @field:Pattern(regexp = "^[0-9]{10}$")
        val businessRegistrationNumber: String,

        @field:NotNull
        @field:PastOrPresent
        val startDate: LocalDate,

        @field:NotBlank
        @field:Size(max = 30)
        val ownerName: String
    )

    data class RegistrationApiForm(
        val businesses: List<Map<String, String>>
    ) {
        companion object {
            fun from(registrationInfo: RegistrationInfo): RegistrationApiForm {
                return RegistrationApiForm(
                    listOf(
                        mapOf(
                            "b_no" to registrationInfo.businessRegistrationNumber,
                            "start_dt" to registrationInfo.startDate.format(DateTimeFormatter.ofPattern("yyyyMMdd")),
                            "p_nm" to registrationInfo.ownerName
                        )
                    )
                )
            }
        }
    }
}